contracts/manager/DelegatedManager.sol

```diff
@@ -0,0 +1,468 @@
+        _;
+    }
+
+    /* ============ State Variables ============ */
+
+    // Instance of SetToken
+    ISetToken public immutable setToken;
+
+    // Address of factory contract used to deploy contract
+    address public immutable factory;
+
+    // Mapping to check if extension is enabled
+    mapping(address => ExtensionState) public extensionAllowlist;
+
+    // Array of enabled extensions
+    address[] internal extensions;
+
+    // Mapping indicating if address is an approved operator
+    mapping(address=>bool) public operatorAllowlist;
+
+    // List of approved operators
+    address[] internal operators;
+
+    // Mapping indicating if asset is approved to be traded for, wrapped into, claimed, etc.
+    mapping(address=>bool) public assetAllowlist;
+
+    // List of allowed assets
+    address[] internal allowedAssets;
+
+    // Toggle if asset allow list is being enforced
+    bool public useAssetAllowlist;
+
+    // Global owner fee split that can be referenced by Extensions
+    uint256 public ownerFeeSplit;
+
+    // Address owners portions of fees get sent to
+    address public ownerFeeRecipient;
+
+    // Address of methodologist which serves as providing methodology for the index and receives fee splits
+    address public methodologist;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        ISetToken _setToken,
+        address _factory,
+        address _methodologist,
+        address[] memory _extensions,
+        address[] memory _operators,
+        address[] memory _allowedAssets,
+        bool _useAssetAllowlist
+    )
+        public
+    {
+        setToken = _setToken;
+        factory = _factory;
+        methodologist = _methodologist;
+        useAssetAllowlist = _useAssetAllowlist;
+
+        _addExtensions(_extensions);
+        _addOperators(_operators);
+        _addAllowedAssets(_allowedAssets);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ONLY EXTENSION: Interact with a module registered on the SetToken.
+     *
+     * @param _module           Module to interact with
+     * @param _data             Byte data of function to call in module
+     */
+    function interactManager(address _module, bytes calldata _data) external onlyExtension {
+        require(_module != address(setToken), "Extensions cannot call SetToken");
+        // Invoke call to module, assume value will always be 0
+        _module.functionCallWithValue(_data, 0);
+    }
+
+    /**
+     * OPERATOR ONLY: Transfers _tokens held by the manager to _destination. Can be used to
+     * distribute fees or recover anything sent here accidentally.
+     *
+     * @param _token           ERC20 token to send
+     * @param _destination     Address receiving the tokens
+     * @param _amount          Quantity of tokens to send
+     */
+    function transferTokens(address _token, address _destination, uint256 _amount) external onlyExtension {
```

Comment:
Think this should be `onlyOwner`

Reply:
Well we want extensions to be able to transfer calculated fee splits

Reply:
Ok makes sense, I thought this was an emergency sweeper or something.